web/hook.go

```diff
@@ -95,6 +178,52 @@ func Hook(c *gin.Context) {
 	})
 }
 
+func getConfigAndMaintainers(c *gin.Context, user *model.User, repo *model.Repo) (*model.Config, *model.Maintainer, error) {
+	rcfile, _ := remote.GetContents(c, user, repo, ".lgtm")
+	config, err := model.ParseConfig(rcfile)
+	if err != nil {
+		log.Errorf("Error parsing .lgtm file for %s. %s", repo.Slug, err)
+		c.String(500, "Error parsing .lgtm file. %s.", err)
+		return nil, nil, err
+	}
+
+	// THIS IS COMPLETELY DUPLICATED IN THE API SECTION. NOT IDEAL
+	file, err := remote.GetContents(c, user, repo, "MAINTAINERS")
+	if err != nil {
+		log.Debugf("no MAINTAINERS file for %s. Checking for team members.", repo.Slug)
+		members, merr := cache.GetMembers(c, user, repo.Owner)
+		if merr != nil {
+			log.Errorf("Error getting repository %s. %s", repo.Slug, err)
+			log.Errorf("Error getting org members %s. %s", repo.Owner, merr)
+			c.String(404, "MAINTAINERS file not found. %s", err)
+			return nil, nil, err
+		} else {
+			for _, member := range members {
+				file = append(file, member.Login...)
+				file = append(file, '\n')
+			}
+		}
+	}
+
+	maintainer, err := model.ParseMaintainer(file)
+	if err != nil {
+		log.Errorf("Error parsing MAINTAINERS file for %s. %s", repo.Slug, err)
+		c.String(500, "Error parsing MAINTAINERS file. %s.", err)
+		return nil, nil, err
+	}
+	return config, maintainer, nil
+}
+
+func getComments(c *gin.Context, user *model.User, repo *model.Repo, num int) ([]*model.Comment, error) {
+	comments, err := remote.GetComments(c, user, repo, num)
+	if err != nil {
```

Comment:
not sure this needs to be a separate function since it is just logging / writing the response. In general I'd like to avoid writing to the response in the helper function, as a convention. It make it easier to trace who/what is writing to the request. Also makes it easier to add some unit tests since we don't have to mock the request